src/main/java/apoc/export/csv/CsvFormat.java

```diff
@@ -120,6 +129,111 @@ public void writeAll(SubGraph graph, Reporter reporter, ExportConfig config, CSV
+                            switch (s) {
+                                case ":START_ID":
+                                    return String.valueOf(r.getStartNodeId());
+                                case ":END_ID":
+                                    return String.valueOf(r.getEndNodeId());
+                                case ":TYPE":
+                                    return entrySet.getKey().name();
+                                default:
+                                    String prop = s.split(":")[0];
+                                    return prop.equals("id") ? String.valueOf(r.getId()) : cleanPoint(FormatUtils.toString(r.getProperty(prop, "")));
+                            }
+                        }).collect(Collectors.toList());
+                    })
+                    .collect(Collectors.toList());
+            writeRow(config, writer, headerRel, rows, "relationships." + entrySet.getKey().name());
+        });
+    }
+
+    private String cleanPoint(String point) {
+        point = point.replace(",\"z\":null", "");
+        point = point.replace(",\"heigth\":null", "");
+        point = point.replace("\"", "");
+        return point;
+    }
+
+    private Set<String> generateHeaderNode(Map.Entry<Iterable<Label>, List<Node>> entrySet) {
+        Set<String> headerNode = new LinkedHashSet<>();
+        headerNode.add("id:ID");
+        Map<String,Class> keyTypes = new LinkedHashMap<>();
+        entrySet.getValue().forEach(node -> updateKeyTypes(keyTypes, node));
+        headerNode.addAll(keyTypes.entrySet().stream().map(stringClassEntry -> formatHeader(stringClassEntry)).collect(Collectors.toCollection(LinkedHashSet::new)));
+        headerNode.add(":LABEL");
+        return headerNode;
+    }
+
+    private Set<String> generateHeaderRelationship(Map.Entry<RelationshipType, List<Relationship>> entrySet) {
+        Set<String> headerNode = new LinkedHashSet<>();
+        Map<String,Class> keyTypes = new LinkedHashMap<>();
+        entrySet.getValue().forEach(relationship -> updateKeyTypes(keyTypes, relationship));
+        headerNode.addAll(keyTypes.entrySet().stream().map(stringClassEntry -> formatHeader(stringClassEntry)).collect(Collectors.toCollection(LinkedHashSet::new)));
+        headerNode.add(":START_ID");
+        headerNode.add(":END_ID");
+        headerNode.add(":TYPE");
+        return headerNode;
+    }
+
+    private void writeRow(ExportConfig config, FileManagerFactory.ExportCypherFileManager writer, Set<String> headerNode, List<List<String>> rows, String name) {
+        try (PrintWriter pw = writer.getPrintWriter(name);
+             CSVWriter csvWriter = getCsvWriter(pw, config)) {
+            if (config.isSeparateHeader()) {
+                try (PrintWriter pwHeader = writer.getPrintWriter("header." + name)) {
+                    CSVWriter csvWriterHeader = getCsvWriter(pwHeader, config);
+                    csvWriterHeader.writeNext(headerNode.toArray(new String[0]), false);
```

Comment:
can we use a reusable string array of the correct size for all these writeNext? in this method